### ISSUE TITLE
compose 1.5.0rc1 fails to work on something that works with 1.4.2

### ISSUE THREAD

**Issue by SvenDowideit** [docker/compose#2269]
see https://github.com/docker/docs.docker.com/tree/hugo-github-linking

if you checkout the `hugo-github-linking` branch, and run with 1.5.0rc1 you get:

```
15:26 $ make clean
docker-compose -p docsdockercom rm -fv ; \
docker rmi $( docker images | grep -E '^docsdockercom_' | awk '{print $1}' ) 2>/dev/null ||:
The AWS_S3_BUCKET variable is not set. Substituting a blank string.
Invalid interpolation format for "command" option in service "cleanup": "/bin/bash -c ' \
if [[ "$AWS_S3_BUCKET" =~ "/" ]] ; then
    BUCKET_PATH=$( echo "$AWS_S3_BUCKET" | sed "s/[^\/]*\///" )
    BUCKET_PATH+="/"
    AWS_S3_BUCKET=$( echo "$AWS_S3_BUCKET" | sed "s/\/.*//")
else
    BUCKET_PATH=
fi

[ -z "$RM_OLDER_THAN" ] && exit 1
CUTOFF_UNIX_TS=$( date --date "$RM_OLDER_THAN" '+%s' )
aws s3 ls --recursive s3://$AWS_S3_BUCKET/$BUCKET_PATH | while read -a LINE ; do
    DATE="${LINE[0]}"
    TIME="${LINE[1]}"
    SIZE="${LINE[2]}"
    NAME="${LINE[*]:3}"

    VERSION_REGEX="^${BUCKET_PATH}v[0-9]+\.[0-9]+/"
    UNIX_TS=$( date --date "$DATE $TIME" "+%s" )

    if [[ "$NAME" =~ $VERSION_REGEX ]] || [[ "$CUTOFF_UNIX_TS" -le "$UNIX_TS" ]] ; then
        echo "Keeping $NAME"
        continue
    fi

    echo "Creating redirect for $NAME"
    aws s3 cp "s3://$AWS_S3_BUCKET/$NAME" "s3://$AWS_S3_BUCKET/$NAME" --website-redirect="/${BUCKET_PATH}index.html" --acl=public-read > /dev/null
done
'
"
docker rmi -f docsdockercom:latest 2>/dev/null ||:
✔ ~/src/docs.docker.com [hugo-github-linking|…5⚑ 3] 
```

when using compose 1.4.2, you get:

```
15:29 $ make clean
docker-compose -p docsdockercom rm -fv ; \
docker rmi $( docker images | grep -E '^docsdockercom_' | awk '{print $1}' ) 2>/dev/null ||:
No stopped containers
docker rmi -f docsdockercom:latest 2>/dev/null ||:
✔ ~/src/docs.docker.com [hugo-github-linking|…5⚑ 3] 
```

**Comment by mnowster** [comment]
Compose 1.5 includes variable interpolation, it's trying to interpolate `$AWS_S3_BUCKET`, if I'm reading your compose file right I think you might need a literal dollar sign, `$$AWS_S3_BUCKET`.

This has a bit more info which might help https://github.com/docker/compose/blob/master/docs/compose-file.md#variable-substitution

**Comment by dnephin** [comment]
Yup, there is a warning at the top of the output, but maybe we should make a louder note about this in the release notes.

**Comment by SvenDowideit** [comment]
ok, @dnephin @mnowster if 

```
The AWS_S3_BUCKET variable is not set. Substituting a blank string.
Invalid interpolation format for "command" option in service "cleanup": "/bin/bash -c 
```
 is what you are referring to as a "warning", it needs to be changed to something that looks like a compose generated warning - at this point, it just looks like compose has unintentionally fallen over and scraped its knee.

perhaps something like  `WARNING: The AWS_S3_BUCKET is not set` probably isn't enough, as you're breaking something that has worked in the previous version.

perhaps _not_ printing out the contents by default (save that for a --verbose?) but actually mentioning that you're intentionally breaking backwards compatibility, and a URL explaining why and how to deal with it.

**Comment by dnephin** [comment]
I agree we need to be more clear about this breaking change. I don't see how not printing the offending variable name is a good idea though. A user shouldn't need to add --verbose to find out what they missed.

The first line is the warning, the second line is the failure caused by the breaking change. Not all warnings will result in a failure, but it's definitely not clear what is happening.

I'm going to do the following:
* add a big bold notice in the release notes about this breaking  change
* change the warning message to `WARNING: the AWS_S3_BUCKET environment variable is not set. Defaulting to a blank string`.
* use color (probably yellow for warning?) to make the WARNING stand out
* see about adding ERROR: to the beginning of error messages, so it's clear that the warning causes the error.

**Comment by dnephin** [comment]
@SvenDowideit @dmp42 How does this notice sound: https://github.com/dnephin/compose/blob/bump-1.5.0rc3/CHANGELOG.md#150-2015-11-02

I'll add it to the github release notes for rc2 as well, if it sounds alright.

**Comment by dnephin** [comment]
Proposed fixes in #2288
